### PR TITLE
[RFP] Catch OSError exceptions from subprocess.Popen()

### PR DESCRIPTION
**Pull request by guedou**
This is proposal to prevent Scapy from crashing when calling commands from `conf.prog.*` as described in #333.

### CODE REVIEW

**Comment by gpotter2** [comment]
How about using a Context Manager to do this properly on every call ?

Something like (not tested)

```
SubProcessContextManager(object):
    def __init__(self, name):
        self.name = name
    def __enter__(self):
        pass
    def __exit__(self, exc_type, exc_value, traceback):
        if exc_type == OSError:
            print(name + " failed")
            ........

with SubProcessContextManager("wireshark()"):
    subprocess.Popen([conf.prog.wireshark, "-r", f])
```

**Comment by guedou** [comment]
I don't use them often, but your construct looks way nicer ! @p-l- do you agree on this construct ?

File: scapy/utils.py [hunk 1]
```diff
@@ -1124,7 +1124,11 @@ def wireshark(pktlist):
     """Run wireshark on a list of packets"""
     f = get_temp_file()
     wrpcap(f, pktlist)
-    subprocess.Popen([conf.prog.wireshark, "-r", f])
+    try:
+        subprocess.Popen([conf.prog.wireshark, "-r", f])
+    except OSError as e:
+        msg = "wireshark(): executing '%s' failed with '%s'"
+        log_scapy.error(msg % (conf.prog.wireshark, e))
```
Comment (p-l-) [inline scapy/utils.py]: You should not format the message here but let `.error()` format it for you. Also, you can use `exc_info` to add the exception information to the logged message:

```Python
    except OSError:
        log_scapy.error("wireshark(): executing %r failed", conf.prog.wireshark, exc_info=1)
```
What do you think?

**Comment by p-l-** [comment]
Fixes #333

**Pull request merged**